python-avd/pyavd/_cv/client/studio_topology.py

```diff
@@ -0,0 +1,359 @@
+        """
+        device_inputs_by_id = {device_id: {"hostname": hostname, "macAddress": system_mac} for device_id, hostname, system_mac in device_inputs}
+
+        # We need to get all the devices to make sure we get the correct index of devices.
+        studio_inputs: dict = await self.get_studio_inputs(studio_id=TOPOLOGY_STUDIO_ID, workspace_id=workspace_id, default_value={}, timeout=timeout)
+
+        request = InputsConfigSetSomeRequest(values=[])
+
+        for device_index, device_entry in enumerate(studio_inputs.get("devices", [])):
+            if not isinstance(device_entry, dict):
+                continue
+
+            device_id = str(device_entry.get("tags", {}).get("query", "")).removeprefix("device:")
+
+            # Ignore the device if it is not one of the requested devices.
+            if device_id not in device_inputs_by_id:
+                continue
+
+            # Update the given fields for the device and add a separate SetSome entry for this device.
+            device_info: dict = device_entry.get("inputs", {}).get("device", {})
+            device_info.update(device_inputs_by_id.pop(device_id))
+
+            request.values.append(
+                InputsConfig(
+                    key=InputsKey(
+                        studio_id=TOPOLOGY_STUDIO_ID,
+                        workspace_id=workspace_id,
+                        path=RepeatedString(values=["devices", str(device_index), "inputs", "device"]),
+                    ),
+                    inputs=json.dumps(device_info),
+                ),
+            )
+
+        index_offset = len(studio_inputs.get("devices", []))
+        # Add any devices not part of the topology studio already.
+        for index, device in enumerate(device_inputs_by_id.items()):
+            device_id, device_inputs = device
+            device_index = index + index_offset
+            device_entry = {
+                "inputs": {"device": {**device_inputs, "modelName": "", "interfaces": []}},
+                "tags": {"query": f"device:{device_id}"},
+            }
+            request.values.append(
+                InputsConfig(
+                    key=InputsKey(
+                        studio_id=TOPOLOGY_STUDIO_ID,
+                        workspace_id=workspace_id,
+                        path=RepeatedString(values=["devices", str(device_index)]),
+                    ),
+                    inputs=json.dumps(device_entry),
+                ),
+            )
+
+        client = InputsConfigServiceStub(self._channel)
+        responses = client.set_some(request, metadata=self._metadata, timeout=timeout + len(request.values) * 0.1)
+        return [response.key async for response in responses]
+
+    @GRPCRequestHandler()
+    async def set_topology_decommissions(
+        self: CVClientProtocol,
+        device_ids: list[str],
+        workspace_id: str,
+        timeout: float = DEFAULT_API_TIMEOUT,
+    ) -> list[DecommissionConfig]:
```

Comment:
The code is returning the following:
```suggestion
    ) -> list[DeviceKey]:
```
Expected?